src/helm/proxy/clients/human_task_client.py

```diff
@@ -0,0 +1,148 @@
+        Otherwise, create a new project using the template. Return the Surge AI project ID."""
+
+        def create_surge_project():
+            project = surge.Project.create(
+                name=template.name,
+                instructions=template.instructions,
+                questions=[self._to_surge_question(question) for question in template.questions],
+                num_workers_per_task=template.num_workers,
+            )
+            return {"id": project.id}
+
+        with _surge_cache_lock:
+            project_response, is_cached = self._cache.get({"template": unstructure(template)}, create_surge_project)
+        project_id = project_response["id"]
+        if is_cached:
+            hlog(f"Reusing existing Surge AI project: {project_id}")
+        else:
+            hlog(f"Creating new Surge AI project: {project_id}")
+        return project_id
+
+    def _get_or_create_task(self, project_id: str, fields: Dict[str, str]) -> str:
+        """Get or create a task on Surge AI and return the Surge AI project ID.
+
+        Attempt to find a Surge AI task inside this project for the fields.
+        If one exists, reuse that task. Otherwise, create a new task inside the project using the fields.
+        Return the Surge AI task ID."""
+        project = surge.Project.retrieve(project_id)
+
+        def create_surge_task():
+            tasks = project.create_tasks([fields], launch=False)  # TODO: Make launch parameter configurable
+            if len(tasks) != 1:
+                return RuntimeError(f"Expected one task in Surge response, but got {len(tasks)} tasks")
+            task = tasks[0]
+            return {"id": task.id}
+
+        with _surge_cache_lock:
+            task_response, is_cached = self._cache.get({"project_id": project_id, "fields": fields}, create_surge_task)
+        task_id = task_response["id"]
+        if is_cached:
+            hlog(f"Reusing existing Surge AI task: {task_id}")
+        else:
+            hlog(f"Creating new Surge AI task: {task_id}")
+        return task_id
+
+    def _get_worker_responses(
+        self, task_id: str, questions: List[HumanQuestionTemplate]
+    ) -> List[HumanTaskWorkerResponse]:
+        task = surge.Task.retrieve(task_id)
+        return [
+            HumanTaskWorkerResponse(answers=[task_response.data[question.text] for question in questions])
+            for task_response in task.responses
+        ]
+
+    def get_human_task(self, request: HumanTaskRequest) -> Optional[HumanTaskRequestResult]:
+        """Create a human task on Surge AI and fetch responses from Surge AI if available.
+
+        Returns HumanTaskRequestResult if worker answers are complete, or None otherwise.
+        The intended use is to call it once to create the task, wait a while, and then call it
+        later to fetch answers.
+
+        First, attempt to find a Surge AI project for the template. If one exists, reuse that project.
```

Comment:
Maybe should define the terminology (ours: questions, templates, Surge: project, task) and state more declaratively what the mapping is before talking about the procedure.

Reply:
Added this in the class docstring.